src/dotty/tools/dotc/transform/FirstTransform.scala

```diff
@@ -160,7 +160,16 @@ class FirstTransform extends MiniPhaseTransform with InfoTransformer with Annota
     ast.Trees.flatten(reorderAndComplete(trees)(ctx.withPhase(thisTransformer.next)))
 
   override def transformOther(tree: Tree)(implicit ctx: Context, info: TransformerInfo) = tree match {
-    case tree: Import => EmptyTree
+    case tree: Import =>
```

Comment:
I think it would be better to move the whole check in Typer or PostTyper. FirstTransform is late for this kind of thing, it will not be called at all in IDEs until you build.